### PR TITLE
Generate pages client

### PR DESCRIPTION
**Pull request by hmharvey**
Part of #1157 (and built on top of #1149). This generalizes the `openapi_client_generator` to generate the Pages client as well. This includes support for **three** new endpoints: [`update_pages_site`](https://developer.github.com/v3/repos/pages/#update-information-about-a-pages-site),[`enable_pages_site`](https://developer.github.com/v3/repos/pages/#enable-a-pages-site), and [`disable_pages_site`](https://developer.github.com/v3/repos/pages/#disable-a-pages-site). 🎉 

The only breaking change is we decided to remove the alias'd definitions - which retroactively affects the deployments client as well.

For a more accurate diff of files changed within the Pages PR and not the previous PRs it's built on, look [here](https://github.com/hmharvey/octokit.rb/compare/deployment-spike...generate-pages-client).

TODO:
- [x] fix definitions and testing for the two new methods. 
  - Hoping to update the docs for `update pages` as it requires one of multiple parameters which isn't documented but it won't be incorporated into OpenAPI for now
- [x] looks like general testing for pages with CI as well

### CODE REVIEW

File: lib/octokit/client/pages.rb [hunk 1]
```diff
@@ -1,63 +1,89 @@
...
-      # @see https://developer.github.com/v3/repos/pages/#list-a-specific-pages-build
-      # @example
-      #   Octokit.pages_build("github/developer.github.com", 5472601)
-      def pages_build(repo, id, options = {})
-        opts = ensure_api_media_type(:pages, options)
-        get "#{Repository.path repo}/pages/builds/#{id}", opts
+      # @param build_id [Integer] The ID of the build
+      # @return <Sawyer::Resource> A single pages build
+      # @see https://developer.github.com/v3/repos/pages/#get-a-specific-pages-build
+      def pages_build(repo, build_id, options = {})
```
Comment (tarebyte) [inline lib/octokit/client/pages.rb]: Naming seems a little odd, not sure if it's possible to make it more clear through auto generation.

File: spec/octokit/client/pages_spec.rb [hunk 2]
```diff
@@ -5,56 +5,93 @@
   before do
     Octokit.reset!
     @client = oauth_client
+    @pages_repo =  "#{test_github_login}/#{test_github_login}.github.io"
```
Comment (tarebyte) [inline spec/octokit/client/pages_spec.rb]: Could we make/ensure this repo exists before the tests start? Maybe in an RSpec context?

File: spec/octokit/client/pages_spec.rb [hunk 3]
```diff
@@ -5,56 +5,93 @@
...
 
-  describe ".latest_pages_build", :vcr do
-    it "lists information about the latest page build" do
-      latest_build = @client.latest_pages_build("github/developer.github.com")
-      expect(latest_build.status).to eq("built")
-      assert_requested :get, github_url("/repos/github/developer.github.com/pages/builds/latest")
-    end
-  end # .latest_pages_build
+  context "with build", :vcr do
+      before do
```
Comment (tarebyte) [inline spec/octokit/client/pages_spec.rb]:
```suggestion
    before do
```

**Comment by mattlangan** [comment]
@hmharvey Is the `enable_pages_site` method supposed to be available in the Octokit Ruby wrapper? I see that method available in this PR, but don't see it in the Pages methods on the current release. I'm not sure if I'm losing history between your fork and the original repo or if I'm just not looking at this correctly.

**Comment by stevevancleemput** [comment]
Well I saw that name passing by when I was browsing in my files ...

**Comment by hmharvey** [comment]
👋🏻  @mattlangan this is a separate project and has no bearing on the `4-stable` branch and the current releases.

**Pull request merged**